### PR TITLE
Add keycloak to portal

### PR DESCRIPTION
**Pull request by evans-g-crsj**
### DO NOT MERGE

#[BUG | FEATUER | DOC] [TITLE]

- closes [SKFP-9: Add Keycloak authentication in portal](https://d3b.atlassian.net/browse/SKFP-9)

## Description

Acceptance Criterias

## Validation

- [ ] Code Approved
- [ ] Test Coverage
- [x] QA Done
- [x] Design/UI Approved from design

## QA

https://portal-keycloak-qa.kidsfirstdrc.org/

## Mention

@ QA, Design ...

### CODE REVIEW

**Review by celinepelletier — APPROVED** [review]

File: src/components/UserProfile/HeaderBannerContainer.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ import {
   selectProfile,
 } from 'store/selectors/profile';
 
+import { withApi } from '../../services/api';
```
Comment (francisl) [inline src/components/UserProfile/HeaderBannerContainer.js:15]: ☝️

File: src/components/UserProfile/Settings.js [hunk 2]
```diff
@@ -10,7 +10,6 @@ import { selectUser } from 'store/selectors/users';
 import { SHOW_DELETE_ACCOUNT } from '../../common/constants';
```
Comment (francisl) [inline src/components/UserProfile/Settings.js:10]: ☝️

File: src/components/UserProfile/UserProfilePageContainer.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ import {
 import { selectIsUserAdmin, selectUser } from 'store/selectors/users';
 import { isSelfInUrlWhenLoggedIn } from 'utils';
 
+import { withApi } from '../../services/api';
```
Comment (francisl) [inline src/components/UserProfile/UserProfilePageContainer.js:31]: ☝️

File: src/pages/variantsSearchPage/Suggester.tsx [hunk 4]
```diff
@@ -21,6 +22,8 @@ import {
   selectSuggestions,
 } from 'store/selectors/genomicSuggester';
 
+import { withApi } from '../../services/api';
```
Comment (francisl) [inline src/pages/variantsSearchPage/Suggester.tsx:25]: ☝️

File: src/services/ajax.js [hunk 5]
```diff
@@ -1,19 +1,22 @@
 import axios from 'axios';
 
-import { store } from 'store';
-import { selectUserToken } from 'store/selectors/users';
+import keycloak from '../keycloak';
```
Comment (francisl) [inline src/services/ajax.js:3]: absolute import

File: src/services/members/search.js [hunk 6]
```diff
@@ -1,18 +1,20 @@
-import { initializeApi } from 'services/api';
 import urljoin from 'url-join';
+
 import { personaApiRoot, reactApiSearchMembersApi } from 'common/injectGlobals';
+import { initializeApi } from 'services/api';
+
 import downloader from '../../common/downloader';
+import keycloak from '../../keycloak';
```
Comment (francisl) [inline src/services/members/search.js:8]: ☝️

**Review by francisl — CHANGES_REQUESTED** [review]
Great Job!
I only have one thing. Absolute import!

We choose to enforce it in kids first since it's considered a best practices in nearly all languages including JS.

Absolute import are : easier to refactor, easier to understand, faster to read, less prone to errors

File: src/components/CohortBuilder/VirtualStudiesMenu/LoadQuery.tsx [hunk 7]
```diff
@@ -1,23 +1,26 @@
...
-import '../../LoadShareSaveDeleteQuery/LoadShareSaveDeleteQuery.css';
+import React from 'react';
 import { FolderOpenOutlined } from '@ant-design/icons';
+import { Button, Dropdown, Menu, Tooltip } from 'antd';
 import { MenuClickEventHandler } from 'rc-menu/lib/interface';
+
+import { Api } from 'store/apiTypes';
 import { VirtualStudy } from 'store/virtualStudiesTypes';
 
+import '../../LoadShareSaveDeleteQuery/LoadShareSaveDeleteQuery.css';
```
[the remark this answers is not in the document]
Comment (celinepelletier) [reply src/components/CohortBuilder/VirtualStudiesMenu/LoadQuery.tsx:9]: even for css ?

[the remark this answers is not in the document]
Comment (francisl) [reply src/components/CohortBuilder/VirtualStudiesMenu/LoadQuery.tsx:9]: the good pratice here would be to have a `LoadQuery.css | LoadQuery.module.css | LoadQuery.module.scss` that you load from the source.

if you have to reuse rules, the `module` file should import it

[the remark this answers is not in the document]
Comment (celinepelletier) [reply src/components/CohortBuilder/VirtualStudiesMenu/LoadQuery.tsx]: actually this import is useless, i just remove it

**Pull request merged**